### PR TITLE
feat(now-proto): initial NOW spec implementation

### PR DESCRIPTION
**Pull request by pacmancoder**
Follow-up to https://github.com/Devolutions/devolutions-gateway/pull/912

No changes in the code except directly using `ironrdp-error` and `ironrdp-pdu`

### CODE REVIEW

**Comment by pacmancoder** [comment]
Respective removal PR in `devolutions-gateway` https://github.com/Devolutions/devolutions-gateway/pull/941

File: crates/now-proto-pdu/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+[package]
+name = "now-proto-pdu"
+version = "0.1.0"
+edition = "2021"
+
```
Comment (CBenoit) [inline crates/now-proto-pdu/Cargo.toml:5]: suggestion: Could you move the test into the testsuite and add this?

```rust
[lib]
doctest = false
test = false
```

File: crates/now-proto-pdu/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+[package]
+name = "now-proto-pdu"
+version = "0.1.0"
+edition = "2021"
```
Comment (CBenoit) [inline crates/now-proto-pdu/Cargo.toml:4]: suggestion:
```suggestion
name = "now-proto-pdu"
version = "0.1.0"
readme = "README.md"
description = "NOW protocol PDU encoding and decoding"
edition.workspace = true
license.workspace = true
homepage.workspace = true
repository.workspace = true
authors.workspace = true
keywords.workspace = true
categories.workspace = true
```

File: crates/now-proto-pdu/Cargo.toml
Comment (CBenoit) [inline crates/now-proto-pdu/Cargo.toml:1]: suggestion: Also add this for the lints:

```toml
[lints]
workspace = true
```

File: crates/now-proto-pdu/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+pub fn decode_owned_cursor<T: PduDecodeOwned>(src: &mut ReadCursor<'_>) -> PduResult<T> {
+    T::decode_owned(src)
+}
+
+/// Trait used to produce an owned version of a given PDU.
+pub trait IntoOwnedPdu: Sized {
+    type Owned: 'static;
+
+    fn into_owned_pdu(self) -> Self::Owned;
+}
```
Comment (CBenoit) [inline crates/now-proto-pdu/src/lib.rs:121]: suggestion: Re-use traits from `ironrdp-pdu` crate

**Review by CBenoit — APPROVED** [review]
LGTM! Comments may be addressed in follow-up PRs.

File: crates/now-proto-pdu/src/lib.rs (hunk 3 above)
Comment (pacmancoder) [reply crates/now-proto-pdu/src/lib.rs:121]: Right, completely forgot to remove this, thanks for spotting

Comment (CBenoit) [reply crates/now-proto-pdu/src/lib.rs:121]: Do you want to address this now, or in follow up PRs?

Comment (pacmancoder) [reply crates/now-proto-pdu/src/lib.rs:121]: Let's address this in this PR

**Comment by pacmancoder** [comment]
@CBenoit I updated PR with refactoring commits -- Fixed all issues commented above and fixed all clippy lints after adding workspace-level lints

**Comment by CBenoit** [comment]
@pacmancoder Great job! The workflow ended up being run and green, so I merged it! :rocket:

**Pull request merged**